R/with.R

```diff
@@ -112,18 +113,14 @@ linters_with_tags <- function(tags, ..., packages = "lintr", exclude_tags = "dep
           i = "These are advertised by {.fn available_linters}, but are not exported by package {.pkg {package}}."
         ))
       }
-      linter_factories <- mget(available$linter, envir = pkg_ns)
-      linters <- Map(
-        call_linter_factory,
-        linter_factory = linter_factories,
-        linter_name = names(linter_factories),
-        MoreArgs = list(package = package)
-      )
-      tagged_linters <- c(tagged_linters, linters)
+      for (linter in available$linter) {
```

Comment:
Note to self -- it looks like `delayedAssign()` doesn't snapshot the value of `linter` that exists at this line, instead, when it's actually evaluated, it winds up with the version of `linter` that exists in this environment later, namely, that's `tail(available$linter, 1)`.

So to proceed with this we'd need to do something like `local()` to force the symbol to evaluate as expected later.